tests/osquery/test_api_views.py

```diff
@@ -1091,3 +1102,242 @@ def test_export_distributed_query_results_ok(self):
+        self.assertEqual(response.json(),
+                         [{"id": query.pk,
+                           "name": query.name,
+                           "version": 1,
+                           "compliance_check_enabled": False,
+                           "sql": query.sql,
+                           "minimum_osquery_version": None,
+                           "description": query.description,
+                           "value": '',
+                           "platforms": [],
+                           "created_at": query.created_at.isoformat(),
+                           "updated_at": query.updated_at.isoformat()
+                           }])
+
+    # create queries
+
+    def test_create_query(self):
+        data = {
+            "name": "test_query01",
+            "sql": "select * from osquery_info;",
+            "compliance_check_enabled": False
+        }
+        self.set_permissions("osquery.add_query")
+        response = self.post_json_data(reverse("osquery_api:queries"), data)
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(Query.objects.filter(name='test_query01').count(), 1)
+        query = Query.objects.get(name='test_query01')
+        self.assertEqual(response.json(),
+                         {"id": query.pk,
+                          "name": query.name,
+                          "version": 1,
+                          "compliance_check_enabled": False,
+                          "sql": "select * from osquery_info;",
+                          "minimum_osquery_version": None,
+                          "description": "",
+                          "value": '',
+                          "platforms": [],
+                          "created_at": query.created_at.isoformat(),
+                          "updated_at": query.updated_at.isoformat()
+                          })
+
+    def test_create_query_validate_error(self):
+        data = {
+            "name": get_random_string(12),
+            "sql": "select * from osquery_info;",
+            "compliance_check_enabled": True
+        }
+        self.set_permissions("osquery.add_query")
+        response = self.post_json_data(reverse("osquery_api:queries"), data)
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json(),
+                         {'non_field_errors': ["{compliance_check_enabled: true} only if query contains ztl_status"]})
+
+    def test_create_query_validate_success(self):
+        query_name = get_random_string(12)
+        data = {
+            "name": query_name,
+            "sql": "ztl_status;",
+            "compliance_check_enabled": True
+        }
+        self.set_permissions("osquery.add_query")
+        response = self.post_json_data(reverse("osquery_api:queries"), data)
+        query = Query.objects.get(name=query_name)
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.json()["compliance_check_enabled"], True)
```

Comment:
Only the response is tested, not the presence of the compliance check in the query. `isinstance()` on the query attribute could be used.